### PR TITLE
[C-2495] Refactor left-nav to typescript, update naming, improve perf

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

Various changes to move improve left-nav navigation layout and types.
- Renames NavColumn to LeftNav
- Migrates all .js files to .ts
- Migrates class + connect to hooks where appropriate

### CODE REVIEW

File: packages/web/src/components/nav/desktop/NavButton.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+const { getAccountStatus, getAccountUser } = accountSelectors
+const { resetState: resetUploadState } = uploadActions
+
+export const NavButton = () => {
+  const dispatch = useDispatch()
+  const record = useRecord()
+  const isSignedIn = useSelector((state: AppState) => !!getAccountUser(state))
+  const accountStatus = useSelector(getAccountStatus)
+  const isUploading = useSelector((state: AppState) => state.upload.uploading)
```
Comment (nicoback2) [inline packages/web/src/components/nav/desktop/NavButton.tsx:31]: is there no selector for this?

File: packages/web/src/components/nav/desktop/NavButton.tsx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+  const handleUpload = useCallback(() => {
+    if (!isUploading) {
+      dispatch(resetUploadState())
+    }
+    dispatch(push(UPLOAD_PAGE))
+    record(make(Name.TRACK_UPLOAD_OPEN, { source: 'nav' }))
+  }, [isUploading, dispatch, record])
+
+  console.log(isSignedIn)
```
Comment (nicoback2) [inline packages/web/src/components/nav/desktop/NavButton.tsx:51]: log

File: packages/web/src/components/nav/mobile/ConnectedBottomBar.tsx [hunk 3]
```diff
@@ -32,7 +32,7 @@ type ConnectedBottomBarProps = ReturnType<typeof mapStateToProps> &
   ReturnType<typeof mapDispatchToProps> &
   RouteComponentProps<any>
 
-const ConnectedNavBar = ({
+const ConnectedBottomBar = ({
```
Comment (nicoback2) [inline packages/web/src/components/nav/mobile/ConnectedBottomBar.tsx:35]: lol

**Review by nicoback2 — APPROVED** [review]
lgtm

**Comment by nicoback2** [comment]
@dylanjeffers  For my education - what/where are the main perf improvements here?

**Comment by dylanjeffers** [comment]
> @dylanjeffers For my education - what/where are the main perf improvements here?

hahah omg dont do me like that :D I think all of these changes are relatively minor, but here are my main thoughts:

File: packages/web/src/components/nav/desktop/NavButton.tsx (hunk 1 above)
Comment (dylanjeffers) [reply packages/web/src/components/nav/desktop/NavButton.tsx:31]: no there isnt, happy to add one though

**Pull request merged**